robot/web/web.go

```diff
@@ -735,18 +735,24 @@ func (svc *webService) initAuthHandlers(listenerTCPAddr *net.TCPAddr, options we
 		for _, handler := range options.Auth.Handlers {
 			switch handler.Type {
 			case rpc.CredentialsTypeAPIKey:
-				apiKeys := handler.Config.StringSlice("keys")
-				if len(apiKeys) == 0 {
-					apiKey := handler.Config.String("key")
-					if apiKey == "" {
-						return nil, errors.Errorf("%q handler requires non-empty API key or keys", handler.Type)
-					}
-					apiKeys = []string{apiKey}
+				legacyAPIKeys := parseLegacyAPIKeys(handler)
+				apiKeys := parseAPIKeys(handler)
+				hasLegacyAPIKeys := len(legacyAPIKeys) != 0
+				hasAPIKeys := len(apiKeys) != 0
+
+				switch {
+				case !hasLegacyAPIKeys && !hasAPIKeys:
+					return nil, errors.Errorf("%q handler requires non-empty API key or keys", handler.Type)
+				case hasLegacyAPIKeys && !hasAPIKeys:
+					rpcOpts = append(rpcOpts, rpc.WithAuthHandler(
+						handler.Type,
+						rpc.MakeSimpleMultiAuthHandler(authEntities, legacyAPIKeys),
```

Comment:
i wonder if we should make the names of these handlers include the notion of API keys too? 
or would that be a much larger refactor

Reply:
/ i could also see why we would want to wait until the complete deprecation of secrets to use those because it looks like secrets use those as well

Reply:
since these handlers are defined in `goutils` i would have to update them there. i think we'll want to keep them general for now?